src/sema/mutability.rs

```diff
@@ -358,29 +335,23 @@ fn recurse_statements(stmts: &[Statement], ns: &Namespace, state: &mut StateChec
 
 fn read_expression(expr: &Expression, state: &mut StateCheck) -> bool {
     match expr {
+        Expression::StorageLoad { loc, .. } => {
```

Comment:
When I initially proposed this solution in the standup. I was pointed out that #1544 removed it because it leads to duplicate error messages. And I should try a different solution. But I agree that a `StorageLoad` must always require `view`. So I guess without changing the structure of the tree there is no other solution than tracking the diagnostics. :shrug: